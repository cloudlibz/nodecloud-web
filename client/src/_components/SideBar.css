.ui.sidebar.vertical.left.menu {
  position: fixed;
  left: 0;
  top: 65px;
  height: calc(100vh - 65px) !important;
  box-shadow: none;
  transition: all 300ms ease-in-out;
}
.ui.sidebar.vertical.left.menu.navActive {
  left: 0;
}
