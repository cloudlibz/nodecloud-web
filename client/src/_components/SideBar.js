import React, { Component } from 'react';
import './SideBar.css';
import { history } from '../_helpers/history.js';

class SideBar extends Component {
  constructor(props) {
    super(props);
    this.state = { visible: true };
    this.navActiveState = true;
    this.changeSelectedDashboardService = this.changeSelectedDashboardService.bind(
      this,
    );
  }

  changeSelectedDashboardService(serviceName) {
    history.push(`/${serviceName}`);
    this.props.changeSelectedDashboardService(serviceName);
  }

  render() {
    return (
      <div
        className={
          this.state.visible
            ? 'ui sidebar vertical left menu overlay visible navActive'
            : 'ui sidebar vertical left menu overlay visible'
        }
        id="site-sideBar"
      >
        <div className="ui accordion">
          <a href="#!" className="item" onClick={() => history.push('/home')}>
            <b>Home</b>
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('virtualmachines')}
          >
            Virtual Machines
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('virtualnetworks')}
          >
            Virtual Networks
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('database')}
          >
            Database
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('storage')}
          >
            Storage
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('security')}
          >
            Security
          </a>
        </div>
      </div>
    );
  }
}

export default SideBar;
