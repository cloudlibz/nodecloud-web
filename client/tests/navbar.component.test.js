import TestRenderer from "react-test-renderer";
import SideBar from "../src/_components/NavBar";
import React from "react";

describe("<NavBar /> Unit Tests", () => {
    it("renders correctly", () => {
        var testFunction = jest.fn();
        const testUser = {
            username: 'test'
        }
        const testUserString = JSON.stringify(testUser);
        localStorage.setItem('user', testUserString);
        var component = TestRenderer.create( <
            SideBar handleShowSidebar = {
                testFunction
            }
            />
        );
        expect(component).toMatchSnapshot();
    });
});