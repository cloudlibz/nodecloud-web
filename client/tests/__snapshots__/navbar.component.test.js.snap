// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Unit Tests renders correctly 1`] = `
<div>
  <div
    class="pusher"
  >
    <div
      class="ui menu asd borderless"
    >
      <a
        class="item openbtn"
        href="#!"
        onClick={[Function]}
      >
        <i
          class="icon content"
        />
      </a>
      <img
        alt="Nodecloud logo"
        class="nav-logo"
        onClick={[Function]}
        src="../media/nodecloudlogo.png"
      />
      <a
        class="item"
        href="#!"
        onClick={[Function]}
      >
         
        Node Cloud
      </a>
      <div
        class="right menu"
      >
        <div
          aria-expanded={false}
          className="ui inline dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "padding": 12,
            }
          }
          tabIndex={0}
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            className="default text"
            role="alert"
          >
            Select Provider
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <div
              aria-checked={false}
              aria-selected={true}
              className="selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <img
                className="ui avatar image"
                src="../media/aws.png"
              />
              <span
                className="text"
              >
                Amazon Web Services
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <img
                className="ui avatar image"
                src="../media/azure.png"
              />
              <span
                className="text"
              >
                Azure
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <img
                className="ui avatar image"
                src="../media/gcp.png"
              />
              <span
                className="text"
              >
                Google Cloud Platform
              </span>
            </div>
          </div>
        </div>
        <div
          class="ui dropdown item"
        >
          test
           
          <i
            class="dropdown icon"
          />
          <div
            class="menu"
          >
            <a
              class="item"
              href="#!"
            >
              Help
            </a>
            <a
              class="item"
              href="#!"
            >
              Logout
            </a>
          </div>
        </div>
        <div
          class="item"
        />
      </div>
    </div>
  </div>
</div>
`;
